base/Enums.jl

```diff
@@ -93,7 +93,7 @@ macro enum(T,syms...)
             end
         end
         function Base.show(io::IO,x::$(esc(typename)))
-            if Base.limit_output(io)
+            if get(io, :compact, false)
```

Comment:
I'd rather get an error when a non-`Bool` is passed in the `IOContext` than a silent fallback to `false`. This makes it easier to detect mistakes.